src/badges.js

```diff
@@ -415,56 +419,39 @@ function validateTemplate(template, locations) {
-							location == 'overlay/right/'
-						) {
-							invalidParam.push(
-								`template paramater '${detail}' must be 'overlay', 'callout', 'callout/[name]', 'overlay/left', 'overlay/right', 'overlay/left/[name]' or 'overlay/right/[name]'`
-							);
-						}
-
-						const [L1, L2, L3] = location.split('/');
-						if (L1) {
-							// overlay / callout
-							if (L1 === 'overlay') {
-								// left / right
-								if (L2 === 'left' || L2 === 'right' || !L2) {
-									// allow left/right/undefined
-									// name of overlay
-									if (L3) {
-										const match = locationsBeingUsed.overlay[L2].find((overlay) => overlay.name === L3);
-										if (!match) {
-											invalidParam.push(`template paramater '${detail}' at index ${index} does not match any overlay location in locations.json`);
-										}
-									}
-								} else {
-									invalidParam.push(`template paramater '${detail}' at index ${index} must start with 'overlay', 'overlay/left' or 'overlay/right'`);
+					if (locations?.details) {
+						template.details[detail].map((location, index) => {
+							if (typeof location !== 'string') {
+								invalidParam.push(`template paramater '${detail}' must be an array of strings. Index ${index} is not a string`);
+							} else if (location.includes('/')) {
+								const locationParts = location.split('/');
+								const [section, name] = locationParts;
+								if (locationParts.length > 2) {
+									invalidParam.push(`template paramater '${detail}' at index ${index} must not contain more than one '/'`);
+								}
+								if (location.endsWith('/')) {
+									invalidParam.push(`template paramater '${detail}' at index ${index} must not end with '/'`);
 								}
-							} else if (L1 === 'callout') {
-								// name of callout
-								if (L2) {
-									const match = locationsBeingUsed.callout.find((callout) => callout.name === L2);
-									if (!match) {
-										invalidParam.push(`template paramater '${detail}' at index ${index} does not match any callout location in locations.json`);
-									}
+								if (location.startsWith('/')) {
+									invalidParam.push(`template paramater '${detail}' at index ${index} must not start with '/'`);
 								}
-							} else {
-								invalidParam.push(`template paramater '${detail}' at index ${index} must start with 'overlay' or 'callout'`);
+								if (!section || !['left', 'right', 'callout'].includes(section)) {
+									invalidParam.push(`template paramater '${detail}' at index ${index} must start with 'left/', 'right/' or 'callout/'`);
+								}
+								if (!name) {
+									invalidParam.push(`template paramater '${detail}' at index ${index} must have a name after '/'`);
+								}
+								const match = locations.details[section].find((locationEntry) => locationEntry.tag === name);
+								if (!match) {
+									invalidParam.push(`template paramater '${detail}' at index ${index} does not match any '${section}' location in ${LOCATIONS_FILE}`);
```

Comment:
This locations validation only applies if a local locations.json file exists (line 442)
However, I'm thinking it should also validate the template locations with global locations if that's being used

Reply:
Yea, that is what I was thinking we should be doing. Perhaps it should sync in this order:
1. locations
2. badges

If it fails to sync the locations it should halt syncing. After syncing any possible locations it would then attempt to sync the badges - but first it should pull down the badge locations from the SMC (even if it just synced this).